### PR TITLE
I added sasl.mechanism, and sasl.jaas.config for each cluster. and fix some bugs [#528, #513,#502,#479,#477,#471]

### PR DESCRIPTION
**Pull request by kingmorning**
Hi 
I added sasl.mechanism, and sasl.jaas.config for each cluster. and fix some bugs [#528, #513,#502,#479,#477,#471]
I tested this new functions, mechanism includes PLAIN,GSSAPI,SCRAM-SHA-256,SCRAM-SHA-512.
and sasl.jaas.config is a optional string, such as: "org.apache.kafka.common.security.plain.PlainLoginModule required  username=kafka  password=kakafkaa;"

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by JasonFJ40** [comment]
Thank you @kingmorning!  My SSL issues have been solved!

File: app/controllers/Cluster.scala [hunk 1]
```diff
@@ -114,6 +122,8 @@ class Cluster (val messagesApi: MessagesApi, val kafkaManagerContext: KafkaManag
         )(ClusterTuning.apply)(ClusterTuning.unapply)
       )
       , "securityProtocol" -> nonEmptyText.verifying(validateSecurityProtocol)
+      , "saslMechanism" -> nonEmptyText.verifying(validateSASLmechanism)
```
Comment (patelh) [inline app/controllers/Cluster.scala]: why is this field required?  If I am not using SASL, then why would we want the user to fill this in.

File: app/kafka/manager/actor/cluster/KafkaStateActor.scala [hunk 2]
```diff
@@ -100,7 +101,11 @@ case class KafkaAdminClientActor(config: KafkaAdminClientActorConfig) extends Ba
     }
     props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, config.clusterContext.config.securityProtocol.stringId)
     props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, brokerListStr)
-    log.info(s"Creating admin client with security protocol=${config.clusterContext.config.securityProtocol.stringId} , broker list : $brokerListStr")
+    props.put(SaslConfigs.SASL_MECHANISM, config.clusterContext.config.saslMechanism.stringId)
```
Comment (patelh) [inline app/kafka/manager/actor/cluster/KafkaStateActor.scala]: We should only put sasl properties if cluster is configured for it.

File: app/kafka/manager/KafkaManager.scala [hunk 3]
```diff
@@ -298,6 +302,8 @@ class KafkaManager(akkaConfig: Config) extends Logging {
                     filterConsumers: Boolean,
                     tuning: Option[ClusterTuning],
                     securityProtocol: String,
+                    saslMechanism: String,
```
Comment (patelh) [inline app/kafka/manager/KafkaManager.scala]: This should be optional.  e.g. Option[String]

File: app/kafka/manager/actor/cluster/KafkaStateActor.scala [hunk 4]
```diff
@@ -257,6 +262,10 @@ case class KafkaManagedOffsetCache(clusterContext: ClusterContext
       cp => props.putAll(cp)
     }
     props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, clusterContext.config.securityProtocol.stringId)
+    props.put(SaslConfigs.SASL_MECHANISM, clusterContext.config.saslMechanism.stringId)
+    if (!clusterContext.config.jaasConfig.isEmpty) {
```
Comment (patelh) [inline app/kafka/manager/actor/cluster/KafkaStateActor.scala]: We should only put sasl properties if cluster is configured for it.

File: app/kafka/manager/actor/cluster/KafkaStateActor.scala [hunk 5]
```diff
@@ -1460,6 +1469,10 @@ class KafkaStateActor(config: KafkaStateActorConfig) extends BaseClusterQueryCom
                 val port: Int = broker.endpoints(securityProtocol)
                 consumerProperties.put(BOOTSTRAP_SERVERS_CONFIG, s"${broker.host}:$port")
                 consumerProperties.put(SECURITY_PROTOCOL_CONFIG, securityProtocol.stringId)
+                consumerProperties.put(SaslConfigs.SASL_MECHANISM, kaConfig.clusterContext.config.saslMechanism.stringId)
```
Comment (patelh) [inline app/kafka/manager/actor/cluster/KafkaStateActor.scala]: We should only put sasl properties if cluster is configured for it.

**Review by patelh — CHANGES_REQUESTED** [review]

**Comment by patelh** [comment]
Thanks for your contribution @kingmorning , let's make this change so SASL is optional and not required.

**Comment by patelh** [comment]
If you need help making the changes, please add me as a collaborator to your forked repo and I can make the changes.

**Comment by kingmorning** [comment]
ok，I added you as a collaborator to my forked repo, thank you for this help.
I tried to make the changes, but failed.
I'm not familiar with scala, especially the play framework.

**Review by patelh — APPROVED** [review]

**Comment by patelh** [comment]
@kingmorning Version 1.3.3.21 has these changes, can you test it out on your cluster and then we can do a release.

**Comment by patelh** [comment]
Yes

**Comment by TheMonolithX64** [comment]
@kingmorning @patelh Where can 1.3.3.21 be pulled from to build and try this out in a SSL-enabled cluster?

**Comment by TheMonolithX64** [comment]
I can confirm that build 1.3.3.21+ I can now see Kafka consumers listed in the UI using SSL-only enabled clusters. Tested on multiple secured clusters, two+ nodes w/ SSL client auth required.

**Comment by carlcauchigig** [comment]
Hi, can you take a look at this issue: https://github.com/yahoo/kafka-manager/issues/664 as it might be similar

**Comment by Neustradamus** [comment]
Linked to:
- https://github.com/scram-sasl/info/issues/1

**Pull request merged**